src/constants/files.ts

```diff
@@ -1,2 +1,5 @@
 // eslint-disable-next-line no-control-regex
 export const INVALID_FILENAME_CHARS = /[<>:"/\\|?*\x00-\x1F]/
+
+// Limit the file size for preview to avoid performance issues
+export const FILE_SIZE_PREVIEW_THRESHOLD = 1024 * 1024 * 10
```

Comment:
Could probably be higher, I'll revisit this before opening the PR for review.
We don't want to load large files into the main process memory, so similarly to how GitHub doesn't show previews of large files, we'll display a message if the file size is above this threshold

Reply:
I'd say github case is a bit different, it doesn't show preview to prevent overloading dom and slowing down the browser, while in our case we only show the first 10 rows. We should probably do some testing with large files and see how it affects electron's memory consumption, verify there's no memory leak, and that it gets cleaned up properly. Not part of this PR though 👍